pkg/download/downloader_test.go

```diff
@@ -70,38 +107,82 @@ func Test_extractZIP(t *testing.T) {
 		defer zipReader.Close()
 		stat, _ := zipReader.Stat()
 		if err := extractZIP(tmpDir.Root(), zipReader, stat.Size()); err != nil {
-			t.Fatalf("extractZIP(%s) error = %v", tt.in, err)
+			if !tt.expectErr {
```

Comment:
Are you checking where `tt.expectErr` is set to `true` but `err==nil`?
(This is repeated in several other places.)

It seems like there's a check for this below but I think they can be merged into a single if/else block I guess?

Reply:
there are four conditions to be checked:

1) did the extract fail in a way that wasn't expected?
2) if it didn't fail, was a failure expected?
3a) otherwise the zip part worked; now check whether the expected files appeared; but, 3b) skip this part if it failed as expected

i think there are four paths in the code, matching these four cases.